### PR TITLE
Fix incorrect date selection ranges for history and logbook

### LINKED ISSUE #7147: History's "Last week" option goes one month ahead.
## Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

## The problem

Selecting "last week" under History made the selected dates jump to the next month. 
image

## Expected behavior

"Last week" should have selected 20-26 September, not 20-26 October.

## Steps to reproduce

    1. Navigate to History
    2. Left-click "Start Date"
    3. Left-click "Last week"

## Environment

Version: 0.115.3, installed in a VM on a Windows Server
Experienced in Chrome and Edge

- Home Assistant release with the issue: 0.115.3
- Last working Home Assistant release (if known): Unknown
- Browser and browser version: Chrome  85.0.4183.121; Edge 85.0.564.63
- Operating system: Client: Windows 10. HA: VM on a Windows Server 

## State of relevant entities

N/A

```yaml

```

## Problem-relevant configuration

N/A

```yaml

```

## Javascript errors shown in your browser console/inspector

None

```txt

```

## Additional information

None

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Fix incorrect date selection ranges that can appear if we are either at the beginning or the end of a month where the selection for "This Week" or "Last Week" crosses the month border.

E.g. today (29.12.) the calculation for "This Week" has an end date that lies already in January. Since the same `todayCopy` value is then also used to set "Last Week", we end up in the wrong month. The same can happen at the beginning of a month. Both cases are fixed now for history and logbook.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/frontend/issues/7147
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/logbook/ha-panel-logbook.ts [hunk 1]
```diff
@@ -147,13 +147,13 @@ export class HaPanelLogbook extends LitElement {
     todayEnd.setDate(todayEnd.getDate() + 1);
     todayEnd.setMilliseconds(todayEnd.getMilliseconds() - 1);
 
-    const todayCopy = new Date(today);
-
+    let todayCopy = new Date(today);
     const yesterday = new Date(todayCopy.setDate(today.getDate() - 1));
```
Comment (bramkragten) [inline src/panels/logbook/ha-panel-logbook.ts:151]: I think the real issue would be this line? We should not set the date of `todayCopy`?

Comment (spacegaier) [reply src/panels/logbook/ha-panel-logbook.ts]: Streamlined the coding to get rid of the `todayCopy`.

File: src/panels/history/ha-panel-history.ts [hunk 2]
```diff
@@ -119,28 +119,21 @@ class HaPanelHistory extends LitElement {
...
     thisWeekEnd.setMilliseconds(thisWeekEnd.getMilliseconds() - 1);
 
-    todayCopy = new Date(today); // set again to prevent issues in case thisWeekEnd crossed into next month
-    const lastWeekStart = new Date(
-      todayCopy.setDate(today.getDate() - today.getDay() - 7)
-    );
-    const lastWeekEnd = new Date(
-      todayCopy.setDate(lastWeekStart.getDate() + 7)
-    );
+    const lastWeekStart = new Date(today);
```
Comment (Sobuno) [inline src/panels/history/ha-panel-history.ts:133]: Couldn't this be simplified by basing it on thisWeekStart? Then you can simply do `lastWeekStart.setDate(thisWeekStart.getDate()-7);` on the next line?

File: src/panels/logbook/ha-panel-logbook.ts [hunk 3]
```diff
@@ -147,27 +147,21 @@ export class HaPanelLogbook extends LitElement {
...
 
-    const todayCopy = new Date(today);
-
-    const yesterday = new Date(todayCopy.setDate(today.getDate() - 1));
-    const yesterdayEnd = new Date(yesterday);
-    yesterdayEnd.setDate(yesterdayEnd.getDate() + 1);
+    const yesterday = new Date(today);
+    yesterday.setDate(today.getDate() - 1);
+    const yesterdayEnd = new Date(today);
     yesterdayEnd.setMilliseconds(yesterdayEnd.getMilliseconds() - 1);
```
Comment (bramkragten) [inline src/panels/logbook/ha-panel-logbook.ts:153]: This will give you the current date minus 1 ms? Not yesterday?

Comment (spacegaier) [reply src/panels/logbook/ha-panel-logbook.ts:153]: No, `today` is the start of today, see line 145.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**